analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AsyncVoidMethod.cs

```diff
@@ -33,6 +34,27 @@ private async void EventHandlerCases_MyEvent(object sender, bool e)
         {
             await Task.Run(() => Console.WriteLine("test"));
         }
+
+        static async void OnValueChanged()  // Noncompliant FP, has OnXxx name
+        {
+        }
+
+        static async void Onboard()         // Noncompliant
+        {
+        }
+
+        static async void OnEventNameChanged(BindableObject bindable, object oldValue, object newValue) // Noncompliant FP Xamarin style, has OnXxx name
+        {
+            // Property changed implementation goes here
+        }
+
+        private async void ArbreDesClefs_ItemInvoked(TreeView sender, TreeViewItemInvokedEventArgs args) { }    // Noncompliant FP
+
+
+        // Substitute for reference to Xamarin.Forms, Windows.UI.Xaml.Controls
+        public class BindableObject { }
+        public class TreeView { }
```

Comment:
maybe add also the namespace `Windows.UI.Xaml.Controls` and test with more than TreeView or `TreeViewItemInvokedEventArgs` , and rely also if there are items in that namespace?

although it may be better to keep it simple with `EventArgs` suffix

Reply:
There [A LOT](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls?view=winrt-19041) of types ending with `EventArgs`, so for Xamarin, it should be fine with the suffix. 

I wanted to stay Xamarin-reference-free to have an universal solution. With any class as a sender and any `WhateverNameEndingWithEventArgs` argument (as supported by the old implementation)